stream/src/main/java/org/apache/kafka/clients/processor/Chooser.java

```diff
@@ -15,9 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.kafka.stream;
-
-import org.apache.kafka.stream.internals.RecordQueue;
+package org.apache.kafka.clients.processor;
 
 public interface Chooser {
 
```

Comment:
How come the Chooser needs to take a queue instead of a single record? Also given that the real way you handle prioritization is with a timestamp do we need this as a public interface at all? It seems to me we have too much stuff around prioritization that is all public--TimestampTracker, TimestampExtractor, RecordQueue, Chooser, etc